src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_FlowAnalysis.cs

```diff
@@ -1282,6 +1296,349 @@ static void G()
+        }
+
+        [Fact]
+        public void IdentityConversion_NullCoalescingOperator_04()
+        {
+            var source =
+@"#pragma warning disable 0649
+struct A<T>
+{
+    public static implicit operator B<T>(A<T> a) => default;
+}
+struct B<T>
+{
+    internal T F;
+}
+class C
+{
+    static void F1(A<object>? x, B<object?> y)
+    {
+        (x ?? y).F.ToString();
+    }
+    static void F2(A<object?>? x, B<object> y)
+    {
+        (x ?? y).F.ToString();
+    }
+    static void F3(A<object> x, B<object?>? y)
+    {
+        (y ?? x).F.ToString();
+    }
+    static void F4(A<object?> x, B<object>? y)
+    {
+        (y ?? x).F.ToString();
+    }
+}";
+            var comp = CreateStandardCompilation(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef }, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (14,10): warning CS8619: Nullability of reference types in value of type 'A<object>' doesn't match target type 'B<object?>'.
+                //         (x ?? y).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("A<object>", "B<object?>").WithLocation(14, 10),
+                // (14,9): warning CS8602: Possible dereference of a null reference.
+                //         (x ?? y).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "(x ?? y).F").WithLocation(14, 9),
+                // (18,10): warning CS8619: Nullability of reference types in value of type 'A<object?>' doesn't match target type 'B<object>'.
+                //         (x ?? y).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("A<object?>", "B<object>").WithLocation(18, 10),
+                // (22,15): warning CS8619: Nullability of reference types in value of type 'B<object>' doesn't match target type 'B<object?>'.
+                //         (y ?? x).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("B<object>", "B<object?>").WithLocation(22, 15),
+                // (22,9): warning CS8602: Possible dereference of a null reference.
+                //         (y ?? x).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "(y ?? x).F").WithLocation(22, 9),
+                // (26,15): warning CS8619: Nullability of reference types in value of type 'B<object?>' doesn't match target type 'B<object>'.
+                //         (y ?? x).F.ToString();
+                Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "x").WithArguments("B<object?>", "B<object>").WithLocation(26, 15));
+        }
+
+        [Fact]
+        public void IdentityConversion_NullCoalescingOperator_05()
+        {
+            var source =
+@"class C
+{
+    static void F1(object? x, dynamic? y, dynamic z)
+    {
+        (x ?? y).ToString();
+        (x ?? z).ToString();
```

Comment:
Consider adding comment `// ok`.
Same on `(y ?? z)`

Reply:
Could you update the diagnostics below to reflect the change? Then the comment will be useful ;-)

---
In reply to: [173346524](https://github.com/dotnet/roslyn/pull/25358#discussion_r173346524) [](ancestors = 173346524)